tests/manage/mcg/test_s3_routes.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from time import sleep
+
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.framework.pytest_customization.marks import (
+    bugzilla,
+    skipif_ocs_version,
+    tier3,
+    skipif_external_mode,
+)
+from ocs_ci.ocs import defaults, constants, ocp
+from ocs_ci.ocs.resources.storage_cluster import get_storage_cluster
+
+logger = logging.getLogger(__name__)
+
+RECONCILE_WAIT = 60
+
+
+class TestS3Routes:
+
+    """
+    Tests related to ODF S3 routes
+    """
+
+    @pytest.fixture(scope="function", autouse=True)
+    def revert_routes(self, request):
+        """
+        Teardown function which reverts the routes and storage cluster option back to original.
+        """
+
+        def finalizer():
+            nb_s3_route_obj = ocp.OCP(
+                kind=constants.ROUTE,
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                resource_name="s3",
+            )
+            if (
+                nb_s3_route_obj.data["spec"]["tls"]["insecureEdgeTerminationPolicy"]
+                == "Redirect"
+            ):
+                s3_route_param = '{"spec":{"tls":{"insecureEdgeTerminationPolicy":"Allow","termination":"reencrypt"}}}'
+                nb_s3_route_obj.patch(params=s3_route_param, format_type="merge")
+
```

Comment:
you can verify revert patch has applied to s3 route

Reply:
ack